tensorflow/core/platform/default/device_tracer.cc

```diff
@@ -253,6 +324,44 @@ void CUPTIManager::InternalBufferCompleted(CUcontext ctx, uint32_t streamId,
   port::AlignedFree(buffer);
 }
 
+void CUPTIManager::ActivityCallback(const CUpti_Activity &record) {
+  VLOG(2) << "ActivityCallback " << record.kind;
+
+  switch (record.kind) {
+    case CUPTI_ACTIVITY_KIND_MEMCPY: {
+      auto *memcpy = reinterpret_cast<const CUpti_ActivityMemcpy *>(&record);
+      CUPTIClient* client = correlation_to_client_[memcpy->correlationId];
+      if (!client) {
```

Comment:
Is it okay to skip logging in this case, while not raising any error? How about let the random client save this log? Or deal with it later and write some comment for it?

Reply:
Even though a client is removed, the callback still can be called for the client. If there are 5 clients and 1 client disables its tracing, CUPTI callback must be valid for the rest of 4 clients. However, the callback for the disabled client also called because the client also uses CUDA APIs. In this case, we don't' have to keep logs for it.